imageio/core/functions.py

```diff
@@ -242,17 +242,16 @@ def imwrite(uri, im, format=None, **kwargs):
         Further keyword arguments are passed to the writer. See :func:`.help`
         to see what arguments are available for a particular format.
     """
+    # Don't coerce subclasses as we need to pass the Image
+    im = np.asanyarray(im)
 
     # Test image
-    if isinstance(im, np.ndarray):
-        if im.ndim == 2:
-            pass
-        elif im.ndim == 3 and im.shape[2] in [1, 3, 4]:
-            pass
-        else:
-            raise ValueError("Image must be 2D (grayscale, RGB, or RGBA).")
+    if im.ndim == 2:
+        pass
+    elif im.ndim == 3 and im.shape[2] in [1, 3, 4]:
+        pass
     else:
```

Comment:
Would be nice if this can be reduced to a single `if not (...)`

Reply:
And I think I prefer to fail early here if the dtype is not numeric (or some similar measure).